src/date.rs

```diff
@@ -423,15 +429,8 @@ mod test {
 
     #[test]
     fn random_wide_range() {
-        let max_date = if cfg!(target_pointer_width="32") {
-            // 32bit duration supports only up to u32::MAX seconds
-            i32::MAX as u64
-        } else {
-            // on 64bit we support up to a year 9999
-            253370764800
-        };
-        for _ in 0..10000 {
-            let sec = rand::thread_rng().gen_range(0, max_date);
+        for _ in 0..100000 {
+            let sec = rand::thread_rng().gen_range(0, MAX_SECONDS);
```

Comment:
Small point - since it's exclusive, the upper bound could be `MAX_SECONDS + 1`.

Reply:
Well, I'm hesitating to do that, in case we will have `MAX_SECONDS == u64::MAX`  with some future feature flag. So I've added explicit check for upper bound on every supported platform in f3a2c70.